submitit/snapshot.py

```diff
@@ -0,0 +1,61 @@
+import itertools
+import os
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import List, Optional
+
+
+def run_cmd(str_args, **kwargs):
+    return subprocess.check_output(str_args, **kwargs).decode("utf-8").strip()
+
+
+class SnapshotManager:
+    """Snapshot Manager
+    This class creates a snapshot of the git repository that the script lives in
+    when creating the snapshot.  This is useful for ensuring that remote jobs that
+    get launched don't accidentally pick up unintended local changes.
+
+    Parameters
+    ----------
+    snapshot_dir: Path
+        A path to where the snapshot should be created
+    with_submodules: bool
+        Whether or not submodules should be included in the snapshot
+    exclude: Optional[List[str]]
+        An optional list of patterns to exclude from the snapshot
+
+    Note
+    ----
+    - Only files that are checked in to the repository are included in the snapshot.
+        If you have experimental code that you would like to include in the snapshot,
+        you'll need to `git add` the file first for it to be included
+    """
+
+    def __init__(
+        self, snapshot_dir: Path, with_submodules: bool = False, exclude: Optional[List[str]] = None,
```

Comment:
this could be `exclude: Sequence[str]  = []`

Reply:
Yeah, that's what I had previously, but I get a lint error: `Dangerous default value [] as argument`

Reply:
then, given it's typed as Sequence you could use a tuple (= non-mutation) as default:
`exclude: Sequence[str] = ()`

Reply:
@jrapin, good point!  updated